src/components/form/achievement/AddAchievement.js

```diff
@@ -1,37 +1,129 @@
-import React from 'react';
-import { Close, Trash } from '~/assets/image';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { Formik, Form } from 'formik';
+import React, { useContext } from 'react';
+
 import Button from '~/components/button/Button';
+import * as storage from '~/storage/LocalStorage';
+import { FormContext } from '~/components/FormContext';
 import InputText from '~/components/inputtext/InputText';
 import InputDate from '~/components/inputdate/InputDate';
 import FormHeader from '~/components/formheader/FormHeader';
-import InputTextArea from '~/components/inputtextarea/InputTextArea';
+import * as achievementUtils from '~/utilities/objects/Achievement';
+import validateAchievementInformation from '~/validations/Achievement';
+import OutsideClickDetector from '~/components/detector/OutsideClickDetector';
+
+const AddAchievement = ({ onClose, isEdit, values }) => {
+  const data = useContext(FormContext).data;
+
+  const handleSubmit = formValues => {
+    if (isEdit) {
+      handleSubmitOnEdit(formValues);
+    } else {
+      handleSubmitOnAdd(formValues);
+    }
+
+    data.set(prevState => ({ ...prevState, ...data }));
+
+    storage.saveResume(localStorage, data.get);
```

Comment:
what is this `localStorage`?

Reply:
I have passed directly the built in object

Reply:
okay, I removed it and implemented in the method itself.